libvirt.go

```diff
@@ -449,22 +520,17 @@ func (l *Libvirt) Events(dom string) (<-chan DomainEvent, error) {
 	return c, nil
 }
 
-// Migrate synchronously migrates the domain specified by dom, e.g.,
+// DomainMigrate synchronously migrates the domain specified by dom, e.g.,
 // 'prod-lb-01', to the destination hypervisor specified by dest, e.g.,
 // 'qemu+tcp://example.com/system'. The flags argument determines the
 // type of migration and how it will be performed. For more information
 // on available migration flags and their meaning, see MigrateFlag*.
-func (l *Libvirt) Migrate(dom string, dest string, flags MigrateFlags) error {
+func (l *Libvirt) DomainMigrate(d *Domain, dest string, flags MigrateFlags) error {
```

Comment:
There ought to be a way to accommodate both methods, but it looks like this PR has grown far beyond "minor improvements".  Would you mind filing an issue to discuss your concerns?  That way, some discussion can happen up front and a decision can be agreed upon before more code is written.